controllers/clients.js

```diff
@@ -0,0 +1,121 @@
+const clients = require('../data/clients.json');
+const fs = require('fs');
+
+// Get all Clients
+const getClients = (req, res) => {
+    if (clients.length === 0){
+        return res.status(400).json({ msg: 'Clients not found'});
+    }
+    return res.json(clients);
+}
+
+// Get Clients by Company Type
+const getClientsByCompanyType = (req, res) => {
+    const companyType = req.params.company_type;
+    let filterClients = clients.filter(item => item.company_type.toLowerCase().includes(companyType.toLowerCase()))
+    if (filterClients.length === 0){
+        return res.status(400).json({ msg: `Clients from company type ${companyType} not found`});
+    }
+    return res.json(filterClients);
+}
+
+// Get Client by ID
+const getClientsById = (req, res) => {
+    const id = req.params.id;
+    let filterClientsId = clients.filter(item => item.id == id)
+    if (filterClientsId.length === 0){
+        return res.status(400).json({ msg: `Client with ID ${id} not found`});
```

Comment:
The status code for when the requested resource hasn't been found is `404`.

Also, there's no need to return a message as a json format inside of an object. We could just a string instead:

```suggestion
        return res.status(404).send(`Client with ID ${id} not found`);
```

Reply:
Done!